### ISSUE TITLE
Undocumented/unforeseen breaking change with file preview in Nova 4

### ISSUE THREAD

**Issue by piljac1** [laravel/nova-issues#5586]
- Laravel Version: 8.83.27
- Nova Version: 4.24.4
- PHP Version: 8.0.28
- Database Driver & Version: MySQL 8.0.32
- Operating System and Version: macOS Monterey 12.6.1
- Browser type and version: Google Chrome 113.0.5672.126

### Description:
We have a website where Nova is under a subdomain which looks like this : `https://admin.other-subdomain.example.test`

The front facing URL (`APP_URL`) looks like the following : `https://other-subdomain.example.test`

Since our public storage URL is based on the `APP_URL`, a CORS error is encountered when the preview attempts to load, because since Nova 4, it is fetched via an AJAX call instead of rendering the URL directly in an `<img>` tag like in Nova 3. I don't know if there would/should be a way to bring back Nova 3's behavior in that case, but if not, I think it should probably be listed as a breaking change in the upgrade guide.

The CORS error in question : 
```
Access to fetch at 'https://other-subdomain.example.test/storage/img/HhLgMLAqAI8REf3Wqrr2ZZP78asUWGRasTTeiWfA.png' from origin 'https://admin.other-subdomain.example.test' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

### Detailed steps to reproduce the issue on a fresh Nova installation:
1. Have a Nova domain with an additional subdomain
2. Have a public storage disk that points on the front end domain
3. Add an `ImageField` to a resource set to use the `public` storage disk
4. Upload an image and create the resource entry (submit the creation form)
5. Edit the entry that was just created and open the console to see the CORS error.

**Comment by crynobone** [comment]
Requiring correct CORS is needed for preview image, we will update the documentation to cover this.

**Comment by piljac1** [comment]
> Requiring correct CORS is needed for preview image, we will update the documentation to cover this.
> 
> https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image

In some cases, if you don't want to rely on server configs, it can also be fixed by defining a "proxy" storage disk that has the same parameters as the desired disk but with Nova's URL. Or by dynamically setting the storage disk's URL in the `NovaServiceProvider`. However, there could be unwanted edge cases with that approach (haven't tested it yet), so a proper server configuration is always preferred.

**Comment by davidhemphill** [comment]
Duplicate of #5300
